File: tp6/ej6/c/ej.c
#include<stdio.h>

void swapCopia(int a, int b){
  int temp = a;
  a = b;
  b = temp;
}

void swapReferencia(int *a, int *b){
  int temp = (*a);
  (*a) = (*b);
  (*b) = temp;
}

int main(){
  int a = 1;
  int b = 2;
  swapCopia(a,b);
  printf("%d\n",a);
  printf("%d\n",b);

  int c = 1;
  int d = 2;
  swapReferencia(&c,&d);
  printf("%d\n",c);
  printf("%d\n",d);
  return 0;
}

File: tp8/ej6/pascal/ej.pas
program Ej;

procedure SwapCopia(a:integer; b:integer); 
begin
  a := a+b;
  b := a-b;
  a := a-b;
end;

procedure SwapReferencia(var a:integer;var b:integer); 
begin
  a := a+b;
  b := a-b;
  a := a-b;
end;

procedure SwapReferenciaCopia(var a:integer;b:integer); 
begin
  a := a+b;
  b := a-b;
  a := a-b;
end;

var
  a,b,c,d,e,f: integer;

begin
  a := 1;
  b := 2;
  SwapCopia(a,b);
  WriteLn(a);
  Writeln(b);

  c := 1;
  d := 2;
  SwapReferencia(c,d);
  WriteLn(c);
  Writeln(d);

  e := 1;
  f := 2;
  SwapReferenciaCopia(e,f);
  WriteLn(e);
  WriteLn(f);

end.
